### PR TITLE
Add keyclock support to gvametapublish pravega broker

### PR DESCRIPTION
**Pull request by dz0080**
Add `keyclock_file` parameter to gvametapublish pravega broker

**Test**
The following pipeline works with Pravega hosted on SDP, changes are for `gvametapublish` element:
```
gst-launch-1.0 \
filesrc location=${VIDEO_EXAMPLE} ! decodebin ! \
gvadetect model=${DETECTION_MODEL} model_proc=${DETECTION_MODEL_PROC} device=CPU ! queue ! \
gvaclassify model=${VEHICLE_CLASSIFICATION_MODEL} model-proc=${VEHICLE_CLASSIFICATION_MODEL_PROC} device=CPU object-class=vehicle ! queue ! \
gvametaconvert format=json ! \
gvametapublish method=pravega controller-uri=${CONTROLLER_URI} keyclock_file=${KEYCLOCK_FILE} \
auth_enabled=true disable_cert=true scope=${SCOPE} stream=${STREAM} routing-key=device1 ! \
fakesink
```
Writing to multiple streams also works:
```
gst-launch-1.0 \
filesrc location=${VIDEO_EXAMPLE} ! decodebin ! \
gvadetect model=${DETECTION_MODEL} model_proc=${DETECTION_MODEL_PROC} device=CPU ! queue ! \
gvaclassify model=${VEHICLE_CLASSIFICATION_MODEL} model-proc=${VEHICLE_CLASSIFICATION_MODEL_PROC} device=CPU object-class=vehicle ! queue ! \
gvametaconvert format=json ! \
gvametapublish method=pravega controller-uri=${CONTROLLER_URI} keyclock_file=${KEYCLOCK_FILE} \
auth_enabled=true disable_cert=true scope=${SCOPE} stream=${STREAM} routing-key=device1 ! \
gvametapublish method=pravega controller-uri=${CONTROLLER_URI} keyclock_file=${KEYCLOCK_FILE} \
auth_enabled=true disable_cert=true scope=${SCOPE} stream=${STREAM2} routing-key=device2 ! \
fakesink
```

### CODE REVIEW

File: gst/elements/gvametapublish/gvametapublish.cpp [hunk 1]
```diff
@@ -373,6 +382,12 @@ static void gva_meta_publish_class_init(GvaMetaPublishClass *klass) {
                             "Pravega Controller Uri", 
                             "[method= pravega] Controller Uri",
                             DEFAULT_CONTROLLER_URI, prm_flags));
+    g_object_class_install_property(
+        gobject_class, PROP_KEYCLOAK_FILE,
+        g_param_spec_string("keyclock-file",
+                            "Keyclock File",
+                            "Keyclock File",
```
Comment (vangork) [inline gst/elements/gvametapublish/gvametapublish.cpp]: Change to "[method= pravega] Keyclock authentication file",

File: gst/elements/gvametapublish/pravega/gvametapublishpravega.cpp [hunk 2]
```diff
@@ -43,10 +44,10 @@ class GvaMetaPublishPravegaPrivate {
 
     gboolean start() {
         GST_INFO_OBJECT(_base,
-                        "Gvametapushlish params: \n--controller-uri: %s\n --scope: %s\n --stream:%s \n --routing-key: %s\n",
-                        _controller_uri.c_str(), _scope.c_str(), _stream.c_str(), _routing_key.c_str());
+                        "Gvametapushlish params: \n--controller-uri: %s\n--keyclock_file: %s\n --scope: %s\n --stream:%s \n --routing-key: %s\n",
+                        _controller_uri.c_str(), _keyclock_file.c_str(), _scope.c_str(), _stream.c_str(), _routing_key.c_str());
 
-        _stream_manager = create_stream_manager(_controller_uri.c_str(), false, false, false);
+        _stream_manager = create_stream_manager(_controller_uri.c_str(), _keyclock_file.c_str(), true, true);
```
Comment (vangork) [inline gst/elements/gvametapublish/pravega/gvametapublishpravega.cpp]: do not hard code last two param as `true, true`

Comment (dz0080) [reply gst/elements/gvametapublish/pravega/gvametapublishpravega.cpp]: Makes sense, explicitly ask user for all params now.

File: gst/elements/gvametapublish/base/common.hpp [hunk 3]
```diff
@@ -45,3 +48,5 @@ GST_EXPORT const gchar *file_format_to_string(FileFormat format);
 
 GST_EXPORT GType gva_metapublish_file_format_get_type(void);
 #define GST_TYPE_GVA_METAPUBLISH_FILE_FORMAT (gva_metapublish_file_format_get_type())
+
+#define BOOL_TO_STR(b) ((b) ? "true" : "false")
```
Comment (vangork) [inline gst/elements/gvametapublish/base/common.hpp]: add a new line at the end of the file

**Review by vangork — CHANGES_REQUESTED** [review]

**Review by vangork — APPROVED** [review]

**Pull request merged**